### PR TITLE
sys_math_atan OK

### PR DESCRIPTION
**Pull request by Prakxo**

### CODE REVIEW

File: src/code/sys_math_atan.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    0x1ED0, 0x1ED5, 0x1EDB, 0x1EE0, 0x1EE6, 0x1EEB, 0x1EF0, 0x1EF6, 0x1EFB, 0x1F00, 0x1F06, 0x1F0B, 0x1F10, 0x1F16,
+    0x1F1B, 0x1F20, 0x1F26, 0x1F2B, 0x1F30, 0x1F36, 0x1F3B, 0x1F40, 0x1F45, 0x1F4B, 0x1F50, 0x1F55, 0x1F5A, 0x1F60,
+    0x1F65, 0x1F6A, 0x1F6F, 0x1F75, 0x1F7A, 0x1F7F, 0x1F84, 0x1F8A, 0x1F8F, 0x1F94, 0x1F99, 0x1F9E, 0x1FA4, 0x1FA9,
+    0x1FAE, 0x1FB3, 0x1FB8, 0x1FBD, 0x1FC3, 0x1FC8, 0x1FCD, 0x1FD2, 0x1FD7, 0x1FDC, 0x1FE1, 0x1FE6, 0x1FEC, 0x1FF1,
+    0x1FF6, 0x1FFB, 0x2000,
+};
+
+u16 U_GetAtanTable(f32 y, f32 x) {
+    s32 idx;
+    u16 ret = atntable[0x400];
```
Comment (AngheloAlf) [inline src/code/sys_math_atan.c:82]: is 0x400 the last element of the table? If that's the case could you do `ARRAY_COUNT() - 1`, it makes it clear it is trying to access the last element instead

File: src/code/sys_math_atan.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            tbl = -U_GetAtanTable(-y, x);
+        }
+    }
+    return tbl;
+}
+
+f32 atanf_table(f32 x, f32 y) {
+    s16 tbl = atans_table(x, y);
+
+    return 9.58738019108e-05f * tbl;
```
[the remark this answers is not in the document]
Comment (hensldm) [reply src/code/sys_math_atan.c:139]: `BINANG_TO_RAD`

File: src/code/sys_math_atan.c [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    s32 idx;
+    u16 ret = atntable[ARRAY_COUNT(atntable) - 1];
+
+    if (x == 0.0f) {
+        ret = atntable[ARRAY_COUNT(atntable) - 1];
+    } else {
+
+        idx = ((y / x) * 1024.0f) + 0.5f;
+
+        if (idx > 0x400) {
```
Comment (hensldm) [inline src/code/sys_math_atan.c:92]:
```suggestion
        if (idx >= ARRAY_COUNT(atntable)) {
```

File: src/code/sys_math_atan.c [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        } else {
+            if (-x < y) {
+                tbl = U_GetAtanTable(-x, y) + 0x4000;
+            } else {
+                tbl = 0x8000 - U_GetAtanTable(y, -x);
+            }
+        }
+    } else if (x < 0.0f) {
+        if (-y <= -x) {
+            tbl = (U_GetAtanTable(-y, -x) + 0x8000);
```
Comment (hensldm) [inline src/code/sys_math_atan.c:124]:
```suggestion
            tbl = U_GetAtanTable(-y, -x) + 0x8000;
```

**Review by hensldm — COMMENTED** [review]
2 small nits

**Pull request merged**